storage/mysql/sql.go

```diff
@@ -124,5 +128,30 @@ func readTree(r row) (*trillian.Tree, error) {
 		}
 	}
 
+	// We're going to try to interpret PublicKey as storageSettings, but it could be a
+	// public key from a really old tree, or an empty column from a tree created in the
+	// period between Trillian key material being removed and this column being used for
+	// storing settings.
+	buff := bytes.NewBuffer(publicKey)
+	dec := gob.NewDecoder(buff)
+	ss := &storageSettings{}
+	var o *mysqlpb.StorageOptions
+	if err := dec.Decode(ss); err != nil {
+		// If there are no storageSettings then this tree was created before settings
+		// were supported, and thus we have to populate the settings with the oldest
+		// settings for features.
+		o = &mysqlpb.StorageOptions{
```

Comment:
What change do you think that this introduces for old trees? The goal, which I believe I have scored, is that this non-nil value establishes the same behaviour for old trees as before this PR.

Stepping back, this is to ensure that there is always a StorageSettings available that explicitly describes how this tree should be treated at the storage layer. I _could_ set this to be nil, but then it will require code elsewhere to infer how it should be treated from a nil value. Having a central place (i.e. here) that establishes the modality makes it easier to reason about and to test.

Reply:
Addressed in the latest commit.